frontend/app/src/components/dialogs/WatcherDialog.vue

```diff
@@ -0,0 +1,380 @@
+          @click="cancel()"
+        >
+          Close
+        </v-btn>
+      </v-card-actions>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script lang="ts">
+import { Component, Emit, Prop, Vue, Watch } from 'vue-property-decorator';
+import MessageDialog from '@/components/dialogs/MessageDialog.vue';
+import { Watcher, WatcherArgs } from '@/services/types-api';
+import { WatcherType } from '@/services/types-common';
+import { Message } from '@/store/store';
+
+@Component({
+  components: { MessageDialog }
+})
+export default class WatcherDialog extends Vue {
+  @Prop({ required: true })
+  title!: string;
+  @Prop({ required: true })
+  message!: string;
+  @Prop({ type: Boolean, required: true })
+  display!: boolean;
+
+  @Prop({ required: false, default: 'Watcher Value' })
+  watcherValueLabel!: string;
+  @Prop({ required: true, default: null })
+  watcherContentId!: string | number | null;
+  @Prop({ required: false, default: '' })
+  preselectWatcherType!: string;
+  @Prop({ required: false, type: Array, default: () => [] })
+  existingWatchers!: Watcher[];
+
+  watcherType: string | null = null;
+  watcherOperation: string | null = null;
+  watcherValue: string | null = null;
+
+  loadedWatchers: Watcher[] = [];
+  existingWatchersEdit: {
+    [identifier: string]: boolean;
+  } = {};
+
+  // mounted() {
+  //   this.watcherType = this.preselectWatcherType;
+  //   this.loadedWatchers = JSON.parse(JSON.stringify(this.existingWatchers)); // make a non-reactive copy
+
+  //   this.loadedWatchers.forEach(watcher => {
+  //     this.existingWatchersEdit[watcher.identifier] = false;
+  //   });
+  // }
+
+  @Watch('display')
+  onDialogToggle() {
+    if (this.display) {
+      this.watcherType = this.preselectWatcherType;
+      this.loadedWatchers = JSON.parse(JSON.stringify(this.existingWatchers)); // make a non-reactive copy
+      this.loadedWatchers.forEach(watcher => {
+        this.existingWatchersEdit[watcher.identifier] = false;
+      });
+    } else {
+      this.watcherType = null;
+      this.loadedWatchers = [];
+      this.existingWatchersEdit = {};
+    }
+  }
+
+  watcherTypes = [
+    {
+      text: 'Collateralization Ratio Watcher (Maker)',
+      type: 'makervault_collateralization_ratio',
+      value: 'makervault_collateralization_ratio'
+    }
+  ];
+
+  watcherOperations = {
+    makervault_collateralization_ratio: [
+      { op: 'gt', value: 'gt', text: 'greater than' },
+      { op: 'ge', value: 'ge', text: 'greater than or equal to' },
+      { op: 'lt', value: 'lt', text: 'less than' },
+      { op: 'le', value: 'le', text: 'less than or equal to' }
+    ]
+  };
+
+  changeEditMode(identifier: string) {
+    this.existingWatchersEdit = {
+      ...this.existingWatchersEdit,
+      [identifier]: !this.existingWatchersEdit[identifier]
+    };
+  }
+
+  deleteWatcher(watcher: string) {
+    let watcherPayload: string[] = [];
+    watcherPayload.push(watcher);
+
+    const { commit } = this.$store;
```

Comment:
maybe create some actions for the part that does the requests and updates the state?

Reply:
Can you show me an example?